Sources/MongoSwift/MongoClient.swift

```diff
@@ -21,17 +21,35 @@ public struct ClientOptions {
     /// the server's default write concern will be used.
     public let writeConcern: WriteConcern?
 
+    /// Specifies the strategy to use when coding `Date`s to/from BSON.
```

Comment:
hmmm.... I wonder if we should be more vague here actually, and just say this is the date coding strategy that will be used for all DBs/colls created from this client. and say "go read about `CollectionOptions.dateCodingStrategy` for more info". and in the collection spot we can actually explain what the strategy does, and talk about `MongoCollection.CollectionType` etc. then we don't have to duplicate so much across the comments for these strategies on the client and DB. I'm not sure what is best really...

Reply:
The options strings I've updated to say basically what you suggested. The collection options ones I added a bit about their responsibility to ensure the data matches. 

I moved the line about converting between non-Document codable types to the actual strategy definitions, since jazzy will link to them from the docstrings for each of the options. For the same reason, I don't think I need to include a reference to `CollectionOptions` in any of the other options.

I added a bit about how `CollectionType` interacts with the strategies in its docstring.

It's safe to say that once this is merged I should get working on that guide...lol